### PR TITLE
Update RTD to support new users

### PR DESCRIPTION
**Pull request by hinerm**
These changes restructure the introductory sections of the [read the docs](https://ops.scijava.org/en/latest/) pages to get users into Ops use relatively smoothly (hopefully) by tuning the `Getting Started` section to be more of a "quick start" into scripting, while focusing and fleshing out the "how-to guides" section, e.g. by prioritizing finding Ops before calling them and adding missing information on `Functions` and `Inplaces`.

### CODE REVIEW

File: docs/ops/doc/CallingOps.md [hunk 1]
```diff
@@ -1,61 +1,96 @@
...
 ops.op("filter.gauss")
 ```
 
 ## Passing the inputs with `.input()`
 
-With the name defined in the `OpBuilder` call, we can then chain the inputs with the `input()` method.
+With the name established in the `OpBuilder` chain, we can then specify our input(s) with the `.input()` method.
 
-For our gaussian blur, we will pass as inputs our input image `inImage`, and a `double` as our sigma parameter:
+For this Gaussian blur, we have two inputs: `inImage` is the image we want to blur, and a `double` value as our sigma parameter:
```
Comment (gselzer) [inline docs/ops/doc/CallingOps.md:21]:
```suggestion
For this Gaussian blur, we have two inputs: `inImage` is the image we want to blur, and a `double` value as the standard deviation of the smoothing function:
```

What do you think about using the technical term?

File: docs/ops/doc/SearchingForOps.md [hunk 2]
```diff
@@ -96,35 +109,33 @@ Names:
...
-
+```
 print(ops.help("filter.gauss"))
 ```
 
-```
+```text
 filter.gauss:
 	- (input, sigmas, @CONTAINER container1) -> None
 	- (input, sigmas, outOfBounds = null, @CONTAINER container1) -> None
```
Comment (ctrueden) [inline docs/ops/doc/SearchingForOps.md:124]: Not part of this PR I know, but this output mixes `null`s and `None`s... we should make that more consistent.

**Review by ctrueden — APPROVED** [review]
@hinerm This is super great! Much friendlier! :star_struck: 

I pushed some commits with minor adjustments on top of your new prose, and I think we're good to merge!

File: docs/ops/doc/CallingOps.md [hunk 3]
```diff
@@ -1,61 +1,96 @@
...
-When an Op should be executed many times on different inputs, the `OpBuilder` syntax can be modified to return the *Op* instead. Instead of calling the `.compute()` function at the end of our `OpBuilder` call, we can instead call the `.computer()` method to get back the matched Op:
+Calling our Gaussian blur as a *computer* above is great when we have pre-allocated output, but for other scenarios we can request Ops as *functions* or *inplaces*.
+
+*Functions* are used when we want to *create* the final output, indicated by ending the builder with `.apply()`:
+
+```groovy
+var outImage = ops.op("filter.gauss").input(inImage, 2.0).apply()
+```
+
+*Inplaces* are used when we want to destructively modify one of the existing inputs (which is explicitly forbidden by *computers*; a *computer* Op's output should be a different object from all of its inputs). We indicate this by the `mutate#()` method, where the `#` corresponds to the *parameter index* that will be modified:
```
Comment (gselzer) [inline docs/ops/doc/CallingOps.md:63]: Would it be confusing, or clarifying, to note this is 1-indexed?

File: docs/ops/doc/CallingOps.md [hunk 4]
```diff
@@ -1,61 +1,96 @@
...
+gaussOp.compute(inImage, 2.0, outImage1)
+gaussOp.compute(inImage, 5.0, outImage2)
 ```
 
+While we do pass concrete inputs and outputs in this example, they are essentially just being used to reason about the desired *types* - which we'll cover in the next section.
+
 *Note that the default `OpEnvironment` implementations cache Op requests* - this means that repeated `OpBuilder` requests targeting the same action will be faster than the original matching call.
 
-## Additions: Matching with classes
+### Matching with classes
```
Comment (gselzer) [inline docs/ops/doc/CallingOps.md:93]: Is `classes` better or worse than `types`?

**Review by gselzer — COMMENTED** [review]
I see that @ctrueden merged before I got the chance to finish my review, but we can still think about my points if we want! None are really important!

File: docs/ops/doc/CallingOps.md (hunk 4 above)
Comment (ctrueden) [reply docs/ops/doc/CallingOps.md:93]: 3648c07bd304517d3dd16f8fc547b0ebd28e8b72

File: docs/ops/doc/CallingOps.md (hunk 3 above)
Comment (ctrueden) [reply docs/ops/doc/CallingOps.md:63]: 06c20936aeca474ffed93ed2cdbf771bdeaa2f05

File: docs/ops/doc/CallingOps.md (hunk 1 above)
Comment (ctrueden) [reply docs/ops/doc/CallingOps.md:21]: I like the simpler `our sigma parameter` :wink:

**Comment by ctrueden** [comment]
Sorry for the overlap @gselzer. I filed #250 to address your comments.
